src/vm/wren_vm.c

```diff
@@ -56,7 +56,15 @@ WrenVM* wrenNewVM(WrenConfiguration* config)
   WrenReallocateFn reallocate = defaultReallocate;
   void* userData = NULL;
   if (config != NULL) {
-    reallocate = config->reallocateFn;
+    
+    // ensure config has a reallocate function
+    if (config->reallocateFn != NULL) {
+      reallocate = config->reallocateFn;
+    }
+    else {
+      config->reallocateFn = defaultReallocate;
+    }
```

Comment:
I find the code somewhat strange, requiring looking in the surrounding code to make sure it's doing the right thing. I believe a slight refactoring would make it easier to reason about:
```suggestion
    if (NULL == config->reallocateFn) {
      config->reallocateFn = defaultReallocate;
    }
    reallocate = config->reallocateFn;
```

Reply:
sure, reformatted but with NULL on the other side